src/modules/silentpayments/main_impl.h

```diff
@@ -7,10 +7,313 @@
+
+static int secp256k1_silentpayments_create_output_tweak(secp256k1_scalar *output_tweak_scalar, const unsigned char *shared_secret33, uint32_t k) {
+    secp256k1_sha256 hash;
+    unsigned char hash_ser[32];
+    unsigned char k_serialized[4];
+    int ret, overflow;
+
+    /* Compute hash(shared_secret || ser_32(k))  [sha256 with tag "BIP0352/SharedSecret"] */
+    secp256k1_silentpayments_sha256_init_sharedsecret(&hash);
+    secp256k1_sha256_write(&hash, shared_secret33, 33);
+    secp256k1_write_be32(k_serialized, k);
+    secp256k1_sha256_write(&hash, k_serialized, sizeof(k_serialized));
+    secp256k1_sha256_finalize(&hash, hash_ser);
+    /* Convert output_tweak to a scalar.
+     *
+     * This can only fail if the output of the hash function is zero greater than or equal to the curve order, which
+     * happens with negligible probability. Normally, we would use VERIFY_CHECK as opposed to returning an error
+     * since returning an error here would result in an untestable branch in the code. But in this case, we return
+     * an error to ensure strict compliance with BIP0352.
+     */
+    secp256k1_scalar_set_b32(output_tweak_scalar, hash_ser, &overflow);
+    ret = !secp256k1_scalar_is_zero(output_tweak_scalar);
+    /* Leaking this value would break indistinguishability of the transaction, so clear it. */
+    secp256k1_memclear_explicit(hash_ser, sizeof(hash_ser));
+    secp256k1_sha256_clear(&hash);
+    return ret & !overflow;
+}
+
+static int secp256k1_silentpayments_create_output_pubkey(const secp256k1_context *ctx, secp256k1_xonly_pubkey *output_xonly, const unsigned char *shared_secret33, const secp256k1_pubkey *spend_pubkey, uint32_t k) {
+    secp256k1_ge output_ge;
+    secp256k1_scalar output_tweak_scalar;
+    /* Calculate the output_tweak and convert it to a scalar.
+     *
+     * Note: _create_output_tweak can only fail if the output of the hash function is greater than or equal to the curve order, which is statistically improbable.
+     * Returning an error here results in an untestable branch in the code, but we do this anyways to ensure strict compliance with BIP0352.
+     */
+    if (!secp256k1_silentpayments_create_output_tweak(&output_tweak_scalar, shared_secret33, k)) {
+        return 0;
+    }
+    if (!secp256k1_pubkey_load(ctx, &output_ge, spend_pubkey)) {
+        secp256k1_scalar_clear(&output_tweak_scalar);
+        return 0;
+    }
+    /* `tweak_add` only fails if output_tweak_scalar*G = -spend_pubkey. Considering output_tweak is the output of a hash function,
+     * this will happen only with negligible probability for honestly created spend_pubkey, but we handle this
+     * error anyway to protect against this function being called with a malicious inputs, i.e., spend_pubkey = -(_create_output_tweak(shared_secret33, k))*G
+     */
+    if (!secp256k1_eckey_pubkey_tweak_add(&output_ge, &output_tweak_scalar)) {
+        secp256k1_scalar_clear(&output_tweak_scalar);
+        return 0;
+    };
```

Comment:
In https://github.com/bitcoin-core/secp256k1/commit/ffffd7ff98368b29759cd3d9933896fb9fa69b1f:

Extra semicolon here.